src/provider/store/Store.ts

```diff
@@ -33,8 +33,14 @@ export class Store {
         return this._store.getState() as Immutable<RootState>;
     }
 
-    public dispatch(action: RootAction): void {
-        this._store.dispatch(action);
+    public async dispatch(action: RootAction): Promise<void> {
```

Comment:
Do we want to return the `dispatch` promise/result here? We might want to use it for error handling.